src/api/app/models/local_build_result/for_package.rb

```diff
@@ -0,0 +1,45 @@
+class LocalBuildResult
+  class ForPackage
+    include ActiveModel::Model
+    attr_accessor :package, :project, :show_all
+    attr_reader :excluded_counter, :results
+
+    def initialize(attributes = {})
+      super
+      @excluded_counter = 0
+      buildresults
+    end
+
+    private
+
+    attr_writer :excluded_results_count, :results
+
+    def buildresults
+      @results = {}
+      backend_build_result.each do |result|
+        result.elements('status').each do |status|
+          @results[status['package']] ||= []
+          if excluded?(status['code'])
+            @excluded_counter += 1
+            next
+          end
+          add_local_build_result(result, status)
+        end
+      end
+    end
+
+    def add_local_build_result(result, status)
+      @results[status['package']] << LocalBuildResult.new(repository: result['repository'], architecture: result['arch'],
```

Comment:
[nitpick] I would assign it in the buildresult method
Also you can access the variable with the setter method self.result